src/generic-container.ts

```diff
@@ -87,6 +86,7 @@ export class GenericContainer implements TestContainer {
   private waitStrategy?: WaitStrategy;
   private startupTimeout: Duration = new Duration(60_000, TemporalUnit.MILLISECONDS);
   private useDefaultLogDriver: boolean = false;
+  private authConfig = {} as AuthConfig;
```

Comment:
I think we should define our own type to avoid having to cast. 
Probably it'd be an object with similar fields, and the type definition here would be optional, like 
`private authConfig?: AuthConfig`

Reply:
Corrected